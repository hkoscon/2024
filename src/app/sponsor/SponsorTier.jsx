import PropTypes from 'prop-types';
import SponsorInfo from './SponsorInfo';

import './sponsorTier.scss';

export default function SponsorTier({ name, sponsors }) {
  return (
    <section className="section">
      <h3 className="title sponsorTier__title">{name}</h3>
      {sponsors.map(({
        title, logo, description, links,
      }) => (
        <SponsorInfo
          key={title}
          title={title}
          logo={logo}
          description={description}
          links={links}
        />
      ))}
    </section>
  );
}

SponsorTier.propTypes = {
  name: PropTypes.string.isRequired,
  sponsors: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    logo: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    links: PropTypes.arrayOf(PropTypes.shape({
      title: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
    })).isRequired,
  })).isRequired,
};
