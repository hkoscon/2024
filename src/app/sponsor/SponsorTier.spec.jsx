/* eslint-env jest */

// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import SponsorTier from './SponsorTier';

describe('SponsorTier', () => {
  it('match snapshot', () => {
    const sponsors = [
      {
        title: 'Testing',
        logo: 'sponsor.png',
        description: 'Testing sponsor',
        links: [
          { title: 'Test 1', url: 'http://example.com/test1' },
          { title: 'Test 2', url: 'http://example.com/test2' },
        ],
      },
    ];
    const { container } = render(
      <SponsorTier
        name="Testing Tier"
        sponsors={sponsors}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
