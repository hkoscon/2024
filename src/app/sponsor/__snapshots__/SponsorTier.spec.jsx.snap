// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SponsorTier match snapshot 1`] = `
<div>
  <section
    class="section"
  >
    <h3
      class="title sponsorTier__title"
    >
      Testing Tier
    </h3>
    <div
      class="columns"
    >
      <div
        class="column"
      >
        <div
          class="card"
        >
          <div
            class="card-content"
          >
            <div
              class="content"
            >
              <div
                class="columns sponsorInfo"
              >
                <div
                  class="column is-3"
                >
                  <div
                    class="card-image"
                  >
                    <figure
                      class="image sponsorInfo__logo"
                    >
                      <img
                        alt="Testing"
                        src="sponsor.png"
                      />
                    </figure>
                  </div>
                </div>
                <div
                  class="column is-9"
                >
                  <h4
                    class="title"
                  >
                    Testing
                  </h4>
                  <div
                    class="content"
                  >
                    Testing sponsor
                  </div>
                  <div
                    class="card-footer"
                  >
                    <a
                      class="card-footer-item"
                      href="http://example.com/test1"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Test 1
                    </a>
                    <a
                      class="card-footer-item"
                      href="http://example.com/test2"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Test 2
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
